File: src/main/java/io/curiousoft/ijudi/ordermanagement/repo/PromotionRepo.java
package io.curiousoft.ijudi.ordermanagement.repo;

import io.curiousoft.ijudi.ordermanagement.model.Promotion;
import io.curiousoft.ijudi.ordermanagement.model.StoreType;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Date;
import java.util.List;

public interface PromotionRepo extends MongoRepository<Promotion, String> {

    List<Promotion> findByExpiryDateBefore(Date date);

    List<Promotion> findByExpiryDateAfterAndShopType(Date date, StoreType storeType);

    List<Promotion> findByShopIdAndExpiryDateAfterAndShopType(String storeId, Date date, StoreType storeType);
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/repo/PromotionRepo.kt
package io.curiousoft.izinga.commons.repo

import io.curiousoft.izinga.commons.model.Promotion
import io.curiousoft.izinga.commons.model.StoreType
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.*

interface PromotionRepo : MongoRepository<Promotion?, String?> {
    fun findByExpiryDateBefore(date: Date?): List<Promotion?>?
    fun findByExpiryDateAfterAndShopType(date: Date?, storeType: StoreType?): List<Promotion?>?
    fun findByShopIdAndExpiryDateAfterAndShopType(
        storeId: String?,
        date: Date?,
        storeType: StoreType?
    ): List<Promotion?>?
}